### PR TITLE
[Improvement] Improve role management interface

### LINKED ISSUE #179: [Feature] Improve the role management module
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon-webui/issues) and found nothing similar.

### Motivation

Improve the role management module
re #82

### Solution

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by silenceland**
### Purpose

Close #179

This PR improves the role management interface and add some test case  to fix problems.

### Tests

- `SysRoleControllerTest`

### API and Format

```

/** Obtain role views with pagination. */
@SaCheckPermission("system:role:list")
@GetMapping("/list")
public PageR<SysRole> listRoles(SysRole role) {}

/** Obtain detailed information based on role number. */
@SaCheckPermission("system:role:query")
@GetMapping(value = "/{roleId}")
public R<SysRole> getRole(@PathVariable Integer roleId) {}

/** Add new role. */
@SaCheckPermission("system:role:add")
@PostMapping
public R<Void> add(@Validated @RequestBody SysRole role) {}

/** Update role info. */
@SaCheckPermission("system:role:update")
@PutMapping
public R<Void> update(@Validated @RequestBody SysRole role) {}

/** Update role status. */
@SaCheckPermission("system:role:update")
@PutMapping("/changeStatus")
public R<Void> changeStatus(@RequestBody SysRole role)  {}

/** Delete role. */
@SaCheckPermission("system:role:delete")
@DeleteMapping("/{roleIds}")
public R<Void> delete(@PathVariable Integer[] roleIds)  {}

/** Obtain a list of role selection boxes. */
@SaCheckPermission("system:role:query")
@GetMapping("/all")
public R<List<SysRole>> all()  {}

/** Query the list of assigned user roles. */
@SaCheckPermission("system:role:list")
@GetMapping("/authUser/allocatedList")
public PageR<User> allocatedList(@RequestBody RoleWithUserDTO roleWithUserDTO) {}

/** Query the list of unassigned user roles. */
@SaCheckPermission("system:role:list")
@GetMapping("/authUser/unallocatedList")
public PageR<User> unallocatedList(@RequestBody RoleWithUserDTO roleWithUserDTO) {}

/** Unauthorize User. */
@SaCheckPermission("system:role:update")
@PutMapping("/authUser/cancel")
public R<Void> cancelAuthUser(@RequestBody UserRole userRole) {}

/** Batch Unauthorize User. */
@SaCheckPermission("system:role:update")
@PutMapping("/authUser/cancelAll")
public R<Void> cancelAuthUserAll(Integer roleId, Integer[] userIds) {}

/** Batch Set User Authorization. */
@SaCheckPermission("system:role:update")
@PutMapping("/authUser/selectAll")
public R<Void> selectAuthUserAll(Integer roleId, Integer[] userIds) {}
```

### Documentation

No.

### CODE REVIEW

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/SysRoleController.java [hunk 1]
```diff
@@ -122,37 +124,37 @@ public R<List<SysRole>> all() {
...
-        List<User> list = userService.selectUnallocatedList(user);
+        List<User> list = userService.selectUnallocatedList(roleWithUserDTO);
         return PageR.<User>builder().success(true).total(page.getTotal()).data(list).build();
     }
 
     /** Unauthorize User. */
-    @SaCheckPermission("system:role:edit")
+    @SaCheckPermission("system:role:update")
     @PutMapping("/authUser/cancel")
     public R<Void> cancelAuthUser(@RequestBody UserRole userRole) {
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/SysRoleController.java:145]: Could this interface merge with `cancelAuthUserAll`?

File: paimon-web-server/src/test/java/org/apache/paimon/web/server/controller/SysRoleControllerTest.java [hunk 2]
```diff
@@ -175,4 +179,149 @@ public void testGetRoleList() throws Exception {
...
+                                        .contentType(MediaType.APPLICATION_JSON_VALUE)
+                                        .accept(MediaType.APPLICATION_JSON_VALUE))
+                        .andExpect(MockMvcResultMatchers.status().isOk())
+                        .andDo(MockMvcResultHandlers.print())
+                        .andReturn()
+                        .getResponse()
+                        .getContentAsString();
+
+        R<Void> r = ObjectMapperUtils.fromJSON(responseString, new TypeReference<R<Void>>() {});
+        assertEquals(200, r.getCode());
```
Comment (SteNicholas) [inline paimon-web-server/src/test/java/org/apache/paimon/web/server/controller/SysRoleControllerTest.java:204]: You should assert the data of response to verify whether the response result is expected.

**Review by SteNicholas — CHANGES_REQUESTED** [review]

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/SysRoleController.java (hunk 1 above)
Comment (silenceland) [reply paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/SysRoleController.java:145]: cancelAuthUser used to delete one item by id of the user_role, cancelAuthUserAll used to batch delete by role_id with relation of userIds, I will merge them into one interface

**Comment by silenceland** [comment]
Thanks for your review, I have updated this PR. @SteNicholas PTAL.

File: paimon-web-server/src/test/java/org/apache/paimon/web/server/controller/SysRoleControllerTest.java [hunk 3]
```diff
@@ -134,13 +158,41 @@ public void testEditRole() throws Exception {
...
+                                        .accept(MediaType.APPLICATION_JSON_VALUE))
+                        .andExpect(MockMvcResultMatchers.status().isOk())
+                        .andDo(MockMvcResultHandlers.print())
+                        .andReturn()
+                        .getResponse()
+                        .getContentAsString();
+
+        PageR<?> r = ObjectMapperUtils.fromJSON(responseString, PageR.class);
+        assertNotNull(r);
+        assertTrue(
```
Comment (SteNicholas) [inline paimon-web-server/src/test/java/org/apache/paimon/web/server/controller/SysRoleControllerTest.java:178]: Could you assert the actual data content?

File: paimon-web-server/src/test/java/org/apache/paimon/web/server/controller/SysRoleControllerTest.java [hunk 4]
```diff
@@ -170,9 +250,102 @@ public void testGetRoleList() throws Exception {
 
         PageR<?> r = ObjectMapperUtils.fromJSON(responseString, PageR.class);
         assertNotNull(r);
+
+        assertTrue(
```
Comment (SteNicholas) [inline paimon-web-server/src/test/java/org/apache/paimon/web/server/controller/SysRoleControllerTest.java:254]: Ditto.

**Review by SteNicholas — CHANGES_REQUESTED** [review]

File: paimon-web-server/src/test/java/org/apache/paimon/web/server/controller/SysRoleControllerTest.java (hunk 3 above)
Comment (silenceland) [reply paimon-web-server/src/test/java/org/apache/paimon/web/server/controller/SysRoleControllerTest.java:180]: Thanks for your review, I have updated this PR

File: paimon-web-server/src/test/java/org/apache/paimon/web/server/controller/SysRoleControllerTest.java (hunk 4 above)
Comment (silenceland) [reply paimon-web-server/src/test/java/org/apache/paimon/web/server/controller/SysRoleControllerTest.java:281]: > Ditto.

Thanks for your review, I have updated this PR

File: paimon-web-server/src/test/java/org/apache/paimon/web/server/controller/SysRoleControllerTest.java (hunk 3 above)
Comment (silenceland) [reply paimon-web-server/src/test/java/org/apache/paimon/web/server/controller/SysRoleControllerTest.java:180]: > Could you assert the actual data content?

Thanks for your review, I have updated this PR

**Review by SteNicholas — APPROVED** [review]
LGTM.

**Review by SteNicholas — APPROVED** [review]
LGTM.

**Pull request merged**